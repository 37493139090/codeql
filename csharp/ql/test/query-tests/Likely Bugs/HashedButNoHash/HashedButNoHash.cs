using System.Collections;
using System.Collections.Generic;

public class Test
{
    public void M()
    {
        var h = new Hashtable();
        h.Add(this, null); // $ Alert

        var d = new Dictionary<Test, bool>();
        d.Add(this, false); // $ Alert
    }

    public override bool Equals(object other)
    {
        return false;
    }
}
