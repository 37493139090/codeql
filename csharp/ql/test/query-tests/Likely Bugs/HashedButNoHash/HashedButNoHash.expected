| HashedButNoHash.cs:9:15:9:18 | this access | This expression is hashed, but type 'Test' only defines Equals(...) not GetHashCode(). |
| HashedButNoHash.cs:12:15:12:18 | this access | This expression is hashed, but type 'Test' only defines Equals(...) not GetHashCode(). |
